src/main/java/io/cdap/plugin/gcp/gcs/actions/GCSCopy.java

```diff
@@ -48,7 +48,15 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
   public void run(ActionContext context) throws IOException {
     config.validate(context.getFailureCollector());
 
-    StorageClient storageClient = StorageClient.create(config.getProject(), config.getServiceAccountFilePath());
+    Boolean isServiceAccountFilePath = config.isServiceAccountFilePath();
+    if (isServiceAccountFilePath == null) {
+      context.getFailureCollector().addFailure("Service account type is undefined.",
```

Comment:
General comment applicable to these PRs: let's also add this validation in `configurePipeline`. We can do this as part of `config.validate`.

Reply:
@rmstar 

This check here is added if the value is @macro, because we already have set e default value for "serviceAccountType" in the Widget (the default value is filepath). However, we can remove the @nullable annotation for the property "serviceAccountType" in config in order to make sure all validation criterias are met. Does this still require validation in `config.validate(...)` to be added?

Reply:
I see. No need to add validation in `config.validate()`